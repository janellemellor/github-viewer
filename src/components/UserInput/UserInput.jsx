import React from 'react';
import PropTypes from 'prop-types';

const UserInput = ({ username, onUsernameChange }) => (
  <>
    <input type='text' name='username' value={username} onChange={onUsernameChange} />
  </>
);

UserInput.propTypes = {
  username: PropTypes.string.isRequired,
  onUsernameChange: PropTypes.func.isRequired    
};

export default UserInput;
