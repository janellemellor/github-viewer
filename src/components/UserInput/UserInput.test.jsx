import React from 'react';
import { shallow } from 'enzyme';
import UserInput from './UserInput.jsx';

describe('UserInput', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<UserInput 
      username='janellemellor'
      onUsernameChange={() => {}} 
      onUsernameSubmit={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
