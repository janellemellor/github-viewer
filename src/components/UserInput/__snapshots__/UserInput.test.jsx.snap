// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInput matches a snapshot 1`] = `
<Fragment>
  <input
    name="username"
    onChange={[Function]}
    type="text"
    value="janellemellor"
  />
</Fragment>
`;
